### PR TITLE
feat!: Dart SDK compatibility 

### PR DESCRIPTION
**Pull request by benni-tec**
# Description
This PR makes this package compatible with the Dart SDK. In order to achieve this I changed 3 things:
- Replace `dart:ui`'s Color with a simple ColorData class
- Replaced `Rect` from `flame` with `Rectangle` from `dart:math`
- Replaced `flutter_test` with `test` package

The `ColorData` class is used excatly once for which I prepared a flame_tiled PR (however I need to wait until this is merged right) including conversion extensions.

No `Rectangle` attribute is ever read in `flame_tiled`, evenso `flame` already contains conversion extensions!

## Checklist
- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
  --> `melos run analyze` fails due to `XmlData.value` however I think this requires another issue/PR
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.
  --> I don't think there are any exmaples necessary for this

## Breaking Change
- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

I think this should result in a minor version bump to 0.11.0 to indicate these breaking changes!

In order to achieve dart comaptibilty only two things changed:
- Color: Custom data class --> `flame_tiled` contains a conversion extension
- Rect: Instead using `dart:math`'s Rectangle --> Use `flame`'s `toRect()` method to convert

## Related Issues
- #69

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm!

File: packages/tiled/pubspec.yaml
Comment (benni-tec) [reply packages/tiled/pubspec.yaml]: Fixed 👍

File: packages/tiled/lib/src/common/color.dart
Comment (jtmcdole) [inline packages/tiled/lib/src/common/color.dart:6]: I just saw this in the other PR; is there a mix up between the two? If so; my comment there is valid here as well: why not store the "hex" as one value and do bit shifting in setters?

This is what `Color` was doing before and so the pattern would match and the memory footprint should be the same.

File: packages/tiled/lib/src/common/color.dart
Comment (jtmcdole) [inline packages/tiled/lib/src/common/color.dart:20]: "rgba"? and why deviate from "argb"?

File: packages/tiled/lib/src/common/color.dart
Comment (benni-tec) [reply packages/tiled/lib/src/common/color.dart:6]: Yes you are right that would be more efficient! This change does belong here, but I based the PR‘s branch on this one so I don‘t have to integrate the new types later!

File: packages/tiled/lib/src/common/color.dart
Comment (benni-tec) [reply packages/tiled/lib/src/common/color.dart:20]: I don’t really know, it just felt more natural. For most colors the alpha value is just 0xff so it can have a default value at the end. CSS also puts it at the end #rrggbbaa, I believe for the same reason. 

This constructor is meant for users and not used internally, so I think this way is a better experience. However I‘m happy to change it!

Comment (jtmcdole) [reply packages/tiled/lib/src/common/color.dart:20]: Right, I was trying to stick with the dart:ui color for familiarity.

**Comment by kevmoo** [comment]
Would this help here? https://flutter.dev/go/move-flutter-agnostic-types

If yes, help me understand how.
If not, what's missing?

**Comment by spydon** [comment]
> Would this help here? https://flutter.dev/go/move-flutter-agnostic-types
> 
> If yes, help me understand how.
> If not, what's missing?

Yes! That would be exactly what we need here, it migrates all the types that we need out of flutter.

Thanks!

**Comment by spydon** [comment]
@benni-tec is this ready to be merged?
Because we aren't getting these classes (Color, Rect etc) separated any time soon right @kevmoo?

**Comment by benni-tec** [comment]
I believe it should be ready to merge now, I adjusted Color to fit @jtmcdole's suggestions.

**Review by spydon — APPROVED** [review]
I did some final clean-up so that the analyzer and tests pass (also on non-windows machines), now I think we can finally merge this! :partying_face:

**Comment by kevmoo** [comment]
> @benni-tec is this ready to be merged? Because we aren't getting these classes (Color, Rect etc) separated any time soon right @kevmoo?

That work was on hold until the "big merge" completed. Hoping to jump in again in the new year!

**Pull request merged**
